File: nilay302/Microsoft/Day 5/Number_of_Substrings_Containing_All_Three_Characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int i = 0;
        int j = 0;
        vector<int>a(3,0);
        int ans = 0;
        for(j = 0;j<s.size();j++){
            a[s[j]-'a']++;
            // cout<<a[s[j]-'a']<<" ";
            while(a[0]>0 && a[1]>0 && a[2]>0 && i<=j){
                cout<<"entered"<<" "; 
                ans+=s.size()-j;
                a[s[i]-'a']--;
                i++;
            }
        }
        return ans;
        
    }
};

File: Ravi/Microsoft/Day 5/NumberOfSubstringsContainingAllThreeCharacters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int a = -1;
        int b = -1;
        int c = -1;
        
        int ans = 0;
        
        for(int i = 0; i < s.length(); i++) {
            
            if(s.charAt(i) == 'a') a = i;
            else if(s.charAt(i) == 'b') b = i;
            else c = i;
            
            ans += Math.min(a, Math.min(b, c)) + 1;
        }
        
        return ans;
        
    }
}
